### PR TITLE
Core/Battleground: Rework BattlegroundScore

### PR DESCRIPTION
**Pull request by Golrag**
**Changes proposed:**

-  Remove subclasses of Battleground Score for each battleground zone
-  Use `PVPStat.db2` and `PVPScoreboardLayout.db2`
-  Add another function in Battleground to update PvpStats
- The only thing that was different for each battleground was the values send in PVPMatchPlayerPVPStat

**Issues addressed:**

Closes: None

**Tests performed:**

- Builds
- Tested in game

**Known issues and TODO list:**

- [ ] Some clean up might still be needed

### CODE REVIEW

**Comment by meji46** [comment]
Can't we use PVPScoreboardLayout.db2 and PVPStat.db2?

**Comment by Golrag** [comment]
Eh i wasn't aware of it, lemme check if I can convert it

**Comment by Golrag** [comment]
Updated to use these two DB files

**Review by mdX7 — APPROVED** [review]

File: src/server/game/Battlegrounds/BattlegroundScore.h [hunk 1]
```diff
@@ -30,67 +33,48 @@ enum ScoreType
...
-        uint8 TeamId; // PvPTeamId
-
-        // Default score, present in every type
-        uint32 KillingBlows;
-        uint32 Deaths;
-        uint32 HonorableKills;
-        uint32 BonusHonor;
-        uint32 DamageDone;
-        uint32 HealingDone;
+    BattlegroundScore(ObjectGuid playerGuid, uint32 team, std::unordered_set<uint32> const* pvpStatIds);
```
Comment (BertiRean) [inline src/server/game/Battlegrounds/BattlegroundScore.h:41]: What is the reason to get a pointer and not a const & to pvpStatIds 🤔

File: src/server/game/Battlegrounds/BattlegroundScore.h [hunk 2]
```diff
@@ -30,67 +33,48 @@ enum ScoreType
...
+    virtual ~BattlegroundScore();
+
+    void UpdateScore(uint32 type, uint32 value);
+    void UpdatePvpStat(uint32 pvpStatID, uint32 value);
+
+    // For Logging purpose
+    virtual std::string ToString() const { return ""; }
```
Comment (BertiRean) [inline src/server/game/Battlegrounds/BattlegroundScore.h:48]: Maybe do this a pure function will be better in order to force devs to propose a good String Representation of Score for Logging. @mdX7 What do you think ?

File: src/server/game/DataStores/DB2Stores.cpp [hunk 3]
```diff
@@ -1621,6 +1626,10 @@ uint32 DB2Manager::LoadStores(std::string const& dataPath, LocaleConstant defaul
             sOldContinentsNodesMask[field] |= submask;
     }
 
+    for (PVPScoreboardLayoutEntry const* layout : sPVPScoreboardLayoutStore)
```
Comment (Shauren) [inline src/server/game/DataStores/DB2Stores.cpp:1629]: Why bother loading the Layout db2 when you are only using data from PvPStat?

Comment (Golrag) [reply src/server/game/DataStores/DB2Stores.cpp:1629]: The idea was to only load the ones that are used in a layout. But this is kinda misleading with the name of the property and they have the same amount of records anyway. Will remove it

**Pull request merged**